### PR TITLE
Update QuPath scripts for 0.2.0-m10

### PR DESCRIPTION
**Pull request by melissalinkert**
See #14.

As noted in the commit messages, the import script should remain backwards compatible.  I have tested with m5, m9, and m10 and not seen any issues.

The export script, however, will now only work with m10 due to changes in the annotation/detection stroke thickness properties (https://github.com/qupath/qupath/commit/8e8fdbf).  If <=m9 support is still needed, do not merge this as is.

### CODE REVIEW

**Comment by pwalczysko** [comment]
I can confirm that the export m10 script does work for me in QuPath 0.2.0-m10.

Further, the export script for m9 or lower works for me in QuPath 0.2.0-m8.

Am I right in my observation that in QuPath..m10 with the appropriate script I do get a different stroke and fill properties, whereas with QuPath... m8 I get the same properties for both, which makes it then later appear in OMERO (after importing the OME-XML) as kind of "masks", see screenshot below. If I am not mistaken, then this is a nice additional improvement here.

iviewer, ROIs from QuPath ...m10
Screen Shot 2020-04-29 at 15 03 41

iviewer, ROIs from QuPath ...m8
Screen Shot 2020-04-29 at 15 03 49

**Comment by pwalczysko** [comment]
Regarding the import script, I am having in QuPath...m8 following experience

<details>
  <summary>Click to expand the output of the console in QuPath, see error at the end</summary>

```
INFO: ROI count: 234
INFO: Found 1 annotations for ROI: 0
INFO: Shape count: 1
INFO: ROI 0:0 is a Polygon
INFO: Found 1 annotations for ROI: 1
INFO: Shape count: 1
INFO: ROI 1:0 is a Polygon
INFO: Found 1 annotations for ROI: 2
INFO: Shape count: 1
INFO: ROI 2:0 is a Polygon
INFO: Found 1 annotations for ROI: 3
INFO: Shape count: 1
INFO: ROI 3:0 is a Polygon
INFO: Found 1 annotations for ROI: 4
INFO: Shape count: 1
INFO: ROI 4:0 is a Polygon
INFO: Found 1 annotations for ROI: 5
INFO: Shape count: 1
INFO: ROI 5:0 is a Polygon
INFO: Found 1 annotations for ROI: 6
INFO: Shape count: 1
INFO: ROI 6:0 is a Polygon
INFO: Found 1 annotations for ROI: 7
INFO: Shape count: 1
INFO: ROI 7:0 is a Polygon
INFO: Found 1 annotations for ROI: 8
INFO: Shape count: 1
INFO: ROI 8:0 is a Polygon
INFO: Found 1 annotations for ROI: 9
INFO: Shape count: 1
INFO: ROI 9:0 is a Polygon
INFO: Found 1 annotations for ROI: 10
INFO: Shape count: 1
INFO: ROI 10:0 is a Polygon
INFO: Found 1 annotations for ROI: 11
INFO: Shape count: 1
INFO: ROI 11:0 is a Polygon
INFO: Found 1 annotations for ROI: 12
INFO: Shape count: 1
INFO: ROI 12:0 is a Polygon
INFO: Found 1 annotations for ROI: 13
INFO: Shape count: 1
INFO: ROI 13:0 is a Polygon
INFO: Found 1 annotations for ROI: 14
INFO: Shape count: 1
INFO: ROI 14:0 is a Polygon
INFO: Found 1 annotations for ROI: 15
INFO: Shape count: 1
INFO: ROI 15:0 is a Polygon
INFO: Found 1 annotations for ROI: 16
INFO: Shape count: 1
INFO: ROI 16:0 is a Polygon
INFO: Found 1 annotations for ROI: 17
INFO: Shape count: 1
INFO: ROI 17:0 is a Polygon
INFO: Found 1 annotations for ROI: 18
INFO: Shape count: 1
INFO: ROI 18:0 is a Polygon
INFO: Found 1 annotations for ROI: 19
INFO: Shape count: 1
INFO: ROI 19:0 is a Polygon
INFO: Found 1 annotations for ROI: 20
INFO: Shape count: 1
INFO: ROI 20:0 is a Polygon
INFO: Found 1 annotations for ROI: 21
INFO: Shape count: 1
INFO: ROI 21:0 is a Polygon
INFO: Found 1 annotations for ROI: 22
INFO: Shape count: 1
INFO: ROI 22:0 is a Polygon
INFO: Found 1 annotations for ROI: 23
INFO: Shape count: 1
INFO: ROI 23:0 is a Polygon
INFO: Found 1 annotations for ROI: 24
INFO: Shape count: 1
INFO: ROI 24:0 is a Polygon
INFO: Found 1 annotations for ROI: 25
INFO: Shape count: 1
INFO: ROI 25:0 is a Polygon
INFO: Found 1 annotations for ROI: 26
INFO: Shape count: 1
INFO: ROI 26:0 is a Polygon
INFO: Found 1 annotations for ROI: 27
INFO: Shape count: 1
INFO: ROI 27:0 is a Polygon
INFO: Found 1 annotations for ROI: 28
INFO: Shape count: 1
INFO: ROI 28:0 is a Polygon
INFO: Found 1 annotations for ROI: 29
INFO: Shape count: 1
INFO: ROI 29:0 is a Polygon
INFO: Found 1 annotations for ROI: 30
INFO: Shape count: 1
INFO: ROI 30:0 is a Polygon
INFO: Found 1 annotations for ROI: 31
INFO: Shape count: 1
INFO: ROI 31:0 is a Polygon
INFO: Found 1 annotations for ROI: 32
INFO: Shape count: 1
INFO: ROI 32:0 is a Polygon
INFO: Found 1 annotations for ROI: 33
INFO: Shape count: 1
INFO: ROI 33:0 is a Polygon
INFO: Found 1 annotations for ROI: 34
INFO: Shape count: 1
INFO: ROI 34:0 is a Polygon
INFO: Found 1 annotations for ROI: 35
INFO: Shape count: 1
INFO: ROI 35:0 is a Polygon
INFO: Found 1 annotations for ROI: 36
INFO: Shape count: 1
INFO: ROI 36:0 is a Polygon
INFO: Found 1 annotations for ROI: 37
INFO: Shape count: 1
INFO: ROI 37:0 is a Polygon
INFO: Found 1 annotations for ROI: 38
INFO: Shape count: 1
INFO: ROI 38:0 is a Polygon
INFO: Found 1 annotations for ROI: 39
INFO: Shape count: 1
INFO: ROI 39:0 is a Polygon
INFO: Found 1 annotations for ROI: 40
INFO: Shape count: 1
INFO: ROI 40:0 is a Polygon
INFO: Found 1 annotations for ROI: 41
INFO: Shape count: 1
INFO: ROI 41:0 is a Polygon
INFO: Found 1 annotations for ROI: 42
INFO: Shape count: 1
INFO: ROI 42:0 is a Polygon
INFO: Found 1 annotations for ROI: 43
INFO: Shape count: 1
INFO: ROI 43:0 is a Polygon
INFO: Found 1 annotations for ROI: 44
INFO: Shape count: 1
INFO: ROI 44:0 is a Polygon
INFO: Found 1 annotations for ROI: 45
INFO: Shape count: 1
INFO: ROI 45:0 is a Polygon
INFO: Found 1 annotations for ROI: 46
INFO: Shape count: 1
INFO: ROI 46:0 is a Polygon
INFO: Found 1 annotations for ROI: 47
INFO: Shape count: 1
INFO: ROI 47:0 is a Polygon
INFO: Found 1 annotations for ROI: 48
INFO: Shape count: 1
INFO: ROI 48:0 is a Polygon
INFO: Found 1 annotations for ROI: 49
INFO: Shape count: 1
INFO: ROI 49:0 is a Polygon
INFO: Found 1 annotations for ROI: 50
INFO: Shape count: 1
INFO: ROI 50:0 is a Polygon
INFO: Found 1 annotations for ROI: 51
INFO: Shape count: 1
INFO: ROI 51:0 is a Polygon
INFO: Found 1 annotations for ROI: 52
INFO: Shape count: 1
INFO: ROI 52:0 is a Polygon
INFO: Found 1 annotations for ROI: 53
INFO: Shape count: 1
INFO: ROI 53:0 is a Polygon
INFO: Found 1 annotations for ROI: 54
INFO: Shape count: 1
INFO: ROI 54:0 is a Polygon
INFO: Found 1 annotations for ROI: 55
INFO: Shape count: 1
INFO: ROI 55:0 is a Polygon
INFO: Found 1 annotations for ROI: 56
INFO: Shape count: 1
INFO: ROI 56:0 is a Polygon
INFO: Found 1 annotations for ROI: 57
INFO: Shape count: 1
INFO: ROI 57:0 is a Polygon
INFO: Found 1 annotations for ROI: 58
INFO: Shape count: 1
INFO: ROI 58:0 is a Polygon
INFO: Found 1 annotations for ROI: 59
INFO: Shape count: 1
INFO: ROI 59:0 is a Polygon
INFO: Found 1 annotations for ROI: 60
INFO: Shape count: 1
INFO: ROI 60:0 is a Polygon
INFO: Found 1 annotations for ROI: 61
INFO: Shape count: 1
INFO: ROI 61:0 is a Polygon
INFO: Found 1 annotations for ROI: 62
INFO: Shape count: 1
INFO: ROI 62:0 is a Polygon
INFO: Found 1 annotations for ROI: 63
INFO: Shape count: 1
INFO: ROI 63:0 is a Polygon
INFO: Found 1 annotations for ROI: 64
INFO: Shape count: 1
INFO: ROI 64:0 is a Polygon
INFO: Found 1 annotations for ROI: 65
INFO: Shape count: 1
INFO: ROI 65:0 is a Polygon
INFO: Found 1 annotations for ROI: 66
INFO: Shape count: 1
INFO: ROI 66:0 is a Polygon
INFO: Found 1 annotations for ROI: 67
INFO: Shape count: 1
INFO: ROI 67:0 is a Polygon
INFO: Found 1 annotations for ROI: 68
INFO: Shape count: 1
INFO: ROI 68:0 is a Polygon
INFO: Found 1 annotations for ROI: 69
INFO: Shape count: 1
INFO: ROI 69:0 is a Polygon
INFO: Found 1 annotations for ROI: 70
INFO: Shape count: 1
INFO: ROI 70:0 is a Polygon
INFO: Found 1 annotations for ROI: 71
INFO: Shape count: 1
INFO: ROI 71:0 is a Polygon
INFO: Found 1 annotations for ROI: 72
INFO: Shape count: 1
INFO: ROI 72:0 is a Polygon
INFO: Found 1 annotations for ROI: 73
INFO: Shape count: 1
INFO: ROI 73:0 is a Polygon
INFO: Found 1 annotations for ROI: 74
INFO: Shape count: 1
INFO: ROI 74:0 is a Polygon
INFO: Found 1 annotations for ROI: 75
INFO: Shape count: 1
INFO: ROI 75:0 is a Polygon
INFO: Found 1 annotations for ROI: 76
INFO: Shape count: 1
INFO: ROI 76:0 is a Polygon
INFO: Found 1 annotations for ROI: 77
INFO: Shape count: 1
INFO: ROI 77:0 is a Polygon
INFO: Found 1 annotations for ROI: 78
INFO: Shape count: 1
INFO: ROI 78:0 is a Polygon
INFO: Found 1 annotations for ROI: 79
INFO: Shape count: 1
INFO: ROI 79:0 is a Polygon
INFO: Found 1 annotations for ROI: 80
INFO: Shape count: 1
INFO: ROI 80:0 is a Polygon
INFO: Found 1 annotations for ROI: 81
INFO: Shape count: 1
INFO: ROI 81:0 is a Polygon
INFO: Found 1 annotations for ROI: 82
INFO: Shape count: 1
INFO: ROI 82:0 is a Polygon
INFO: Found 1 annotations for ROI: 83
INFO: Shape count: 1
INFO: ROI 83:0 is a Polygon
INFO: Found 1 annotations for ROI: 84
INFO: Shape count: 1
INFO: ROI 84:0 is a Polygon
INFO: Found 1 annotations for ROI: 85
INFO: Shape count: 1
INFO: ROI 85:0 is a Polygon
INFO: Found 1 annotations for ROI: 86
INFO: Shape count: 1
INFO: ROI 86:0 is a Polygon
INFO: Found 1 annotations for ROI: 87
INFO: Shape count: 1
INFO: ROI 87:0 is a Polygon
INFO: Found 1 annotations for ROI: 88
INFO: Shape count: 1
INFO: ROI 88:0 is a Polygon
INFO: Found 1 annotations for ROI: 89
INFO: Shape count: 1
INFO: ROI 89:0 is a Polygon
INFO: Found 1 annotations for ROI: 90
INFO: Shape count: 1
INFO: ROI 90:0 is a Polygon
INFO: Found 1 annotations for ROI: 91
INFO: Shape count: 1
INFO: ROI 91:0 is a Polygon
INFO: Found 1 annotations for ROI: 92
INFO: Shape count: 1
INFO: ROI 92:0 is a Polygon
INFO: Found 1 annotations for ROI: 93
INFO: Shape count: 1
INFO: ROI 93:0 is a Polygon
INFO: Found 1 annotations for ROI: 94
INFO: Shape count: 1
INFO: ROI 94:0 is a Polygon
INFO: Found 1 annotations for ROI: 95
INFO: Shape count: 1
INFO: ROI 95:0 is a Polygon
INFO: Found 1 annotations for ROI: 96
INFO: Shape count: 1
INFO: ROI 96:0 is a Polygon
INFO: Found 1 annotations for ROI: 97
INFO: Shape count: 1
INFO: ROI 97:0 is a Polygon
INFO: Found 1 annotations for ROI: 98
INFO: Shape count: 1
INFO: ROI 98:0 is a Polygon
INFO: Found 1 annotations for ROI: 99
INFO: Shape count: 1
INFO: ROI 99:0 is a Polygon
INFO: Found 1 annotations for ROI: 100
INFO: Shape count: 1
INFO: ROI 100:0 is a Polygon
INFO: Found 1 annotations for ROI: 101
INFO: Shape count: 1
INFO: ROI 101:0 is a Polygon
INFO: Found 1 annotations for ROI: 102
INFO: Shape count: 1
INFO: ROI 102:0 is a Polygon
INFO: Found 1 annotations for ROI: 103
INFO: Shape count: 1
INFO: ROI 103:0 is a Polygon
INFO: Found 1 annotations for ROI: 104
INFO: Shape count: 1
INFO: ROI 104:0 is a Polygon
INFO: Found 1 annotations for ROI: 105
INFO: Shape count: 1
INFO: ROI 105:0 is a Polygon
INFO: Found 1 annotations for ROI: 106
INFO: Shape count: 1
INFO: ROI 106:0 is a Polygon
INFO: Found 1 annotations for ROI: 107
INFO: Shape count: 1
INFO: ROI 107:0 is a Polygon
INFO: Found 1 annotations for ROI: 108
INFO: Shape count: 1
INFO: ROI 108:0 is a Polygon
INFO: Found 1 annotations for ROI: 109
INFO: Shape count: 1
INFO: ROI 109:0 is a Polygon
INFO: Found 1 annotations for ROI: 110
INFO: Shape count: 1
INFO: ROI 110:0 is a Polygon
INFO: Found 1 annotations for ROI: 111
INFO: Shape count: 1
INFO: ROI 111:0 is a Polygon
INFO: Found 1 annotations for ROI: 112
INFO: Shape count: 1
INFO: ROI 112:0 is a Polygon
INFO: Found 1 annotations for ROI: 113
INFO: Shape count: 1
INFO: ROI 113:0 is a Polygon
INFO: Found 1 annotations for ROI: 114
INFO: Shape count: 1
INFO: ROI 114:0 is a Polygon
INFO: Found 1 annotations for ROI: 115
INFO: Shape count: 1
INFO: ROI 115:0 is a Polygon
INFO: Found 1 annotations for ROI: 116
INFO: Shape count: 1
INFO: ROI 116:0 is a Polygon
INFO: Found 1 annotations for ROI: 117
INFO: Shape count: 1
INFO: ROI 117:0 is a Polygon
INFO: Found 1 annotations for ROI: 118
INFO: Shape count: 1
INFO: ROI 118:0 is a Polygon
INFO: Found 1 annotations for ROI: 119
INFO: Shape count: 1
INFO: ROI 119:0 is a Polygon
INFO: Found 1 annotations for ROI: 120
INFO: Shape count: 1
INFO: ROI 120:0 is a Polygon
INFO: Found 1 annotations for ROI: 121
INFO: Shape count: 1
INFO: ROI 121:0 is a Polygon
INFO: Found 1 annotations for ROI: 122
INFO: Shape count: 1
INFO: ROI 122:0 is a Polygon
INFO: Found 1 annotations for ROI: 123
INFO: Shape count: 1
INFO: ROI 123:0 is a Polygon
INFO: Found 1 annotations for ROI: 124
INFO: Shape count: 1
INFO: ROI 124:0 is a Polygon
INFO: Found 1 annotations for ROI: 125
INFO: Shape count: 1
INFO: ROI 125:0 is a Polygon
INFO: Found 1 annotations for ROI: 126
INFO: Shape count: 1
INFO: ROI 126:0 is a Polygon
INFO: Found 1 annotations for ROI: 127
INFO: Shape count: 1
INFO: ROI 127:0 is a Polygon
INFO: Found 1 annotations for ROI: 128
INFO: Shape count: 1
INFO: ROI 128:0 is a Polygon
INFO: Found 1 annotations for ROI: 129
INFO: Shape count: 1
INFO: ROI 129:0 is a Polygon
INFO: Found 1 annotations for ROI: 130
INFO: Shape count: 1
INFO: ROI 130:0 is a Polygon
INFO: Found 1 annotations for ROI: 131
INFO: Shape count: 1
INFO: ROI 131:0 is a Polygon
INFO: Found 1 annotations for ROI: 132
INFO: Shape count: 1
INFO: ROI 132:0 is a Polygon
INFO: Found 1 annotations for ROI: 133
INFO: Shape count: 1
INFO: ROI 133:0 is a Polygon
INFO: Found 1 annotations for ROI: 134
INFO: Shape count: 1
INFO: ROI 134:0 is a Polygon
INFO: Found 1 annotations for ROI: 135
INFO: Shape count: 1
INFO: ROI 135:0 is a Polygon
INFO: Found 1 annotations for ROI: 136
INFO: Shape count: 1
INFO: ROI 136:0 is a Polygon
INFO: Found 1 annotations for ROI: 137
INFO: Shape count: 1
INFO: ROI 137:0 is a Polygon
INFO: Found 1 annotations for ROI: 138
INFO: Shape count: 1
INFO: ROI 138:0 is a Polygon
INFO: Found 1 annotations for ROI: 139
INFO: Shape count: 1
INFO: ROI 139:0 is a Polygon
INFO: Found 1 annotations for ROI: 140
INFO: Shape count: 1
INFO: ROI 140:0 is a Polygon
INFO: Found 1 annotations for ROI: 141
INFO: Shape count: 1
INFO: ROI 141:0 is a Polygon
INFO: Found 1 annotations for ROI: 142
INFO: Shape count: 1
INFO: ROI 142:0 is a Polygon
INFO: Found 1 annotations for ROI: 143
INFO: Shape count: 1
INFO: ROI 143:0 is a Polygon
INFO: Found 1 annotations for ROI: 144
INFO: Shape count: 1
INFO: ROI 144:0 is a Polygon
INFO: Found 1 annotations for ROI: 145
INFO: Shape count: 1
INFO: ROI 145:0 is a Polygon
INFO: Found 1 annotations for ROI: 146
INFO: Shape count: 1
INFO: ROI 146:0 is a Polygon
INFO: Found 1 annotations for ROI: 147
INFO: Shape count: 1
INFO: ROI 147:0 is a Polygon
INFO: Found 1 annotations for ROI: 148
INFO: Shape count: 1
INFO: ROI 148:0 is a Polygon
INFO: Found 1 annotations for ROI: 149
INFO: Shape count: 1
INFO: ROI 149:0 is a Polygon
INFO: Found 1 annotations for ROI: 150
INFO: Shape count: 1
INFO: ROI 150:0 is a Polygon
INFO: Found 1 annotations for ROI: 151
INFO: Shape count: 1
INFO: ROI 151:0 is a Polygon
INFO: Found 1 annotations for ROI: 152
INFO: Shape count: 1
INFO: ROI 152:0 is a Polygon
INFO: Found 1 annotations for ROI: 153
INFO: Shape count: 1
INFO: ROI 153:0 is a Polygon
INFO: Found 1 annotations for ROI: 154
INFO: Shape count: 1
INFO: ROI 154:0 is a Polygon
INFO: Found 1 annotations for ROI: 155
INFO: Shape count: 1
INFO: ROI 155:0 is a Polygon
INFO: Found 1 annotations for ROI: 156
INFO: Shape count: 1
INFO: ROI 156:0 is a Polygon
INFO: Found 1 annotations for ROI: 157
INFO: Shape count: 1
INFO: ROI 157:0 is a Polygon
INFO: Found 1 annotations for ROI: 158
INFO: Shape count: 1
INFO: ROI 158:0 is a Polygon
INFO: Found 1 annotations for ROI: 159
INFO: Shape count: 1
INFO: ROI 159:0 is a Polygon
INFO: Found 1 annotations for ROI: 160
INFO: Shape count: 1
INFO: ROI 160:0 is a Polygon
INFO: Found 1 annotations for ROI: 161
INFO: Shape count: 1
INFO: ROI 161:0 is a Polygon
INFO: Found 1 annotations for ROI: 162
INFO: Shape count: 1
INFO: ROI 162:0 is a Polygon
INFO: Found 1 annotations for ROI: 163
INFO: Shape count: 1
INFO: ROI 163:0 is a Polygon
INFO: Found 1 annotations for ROI: 164
INFO: Shape count: 1
INFO: ROI 164:0 is a Polygon
INFO: Found 1 annotations for ROI: 165
INFO: Shape count: 1
INFO: ROI 165:0 is a Polygon
INFO: Found 1 annotations for ROI: 166
INFO: Shape count: 1
INFO: ROI 166:0 is a Polygon
INFO: Found 1 annotations for ROI: 167
INFO: Shape count: 1
INFO: ROI 167:0 is a Polygon
INFO: Found 1 annotations for ROI: 168
INFO: Shape count: 1
INFO: ROI 168:0 is a Polygon
INFO: Found 1 annotations for ROI: 169
INFO: Shape count: 1
INFO: ROI 169:0 is a Polygon
INFO: Found 1 annotations for ROI: 170
INFO: Shape count: 1
INFO: ROI 170:0 is a Polygon
INFO: Found 1 annotations for ROI: 171
INFO: Shape count: 1
INFO: ROI 171:0 is a Polygon
INFO: Found 1 annotations for ROI: 172
INFO: Shape count: 1
INFO: ROI 172:0 is a Polygon
INFO: Found 1 annotations for ROI: 173
INFO: Shape count: 1
INFO: ROI 173:0 is a Polygon
INFO: Found 1 annotations for ROI: 174
INFO: Shape count: 1
INFO: ROI 174:0 is a Polygon
INFO: Found 1 annotations for ROI: 175
INFO: Shape count: 1
INFO: ROI 175:0 is a Polygon
INFO: Found 1 annotations for ROI: 176
INFO: Shape count: 1
INFO: ROI 176:0 is a Polygon
INFO: Found 1 annotations for ROI: 177
INFO: Shape count: 1
INFO: ROI 177:0 is a Polygon
INFO: Found 1 annotations for ROI: 178
INFO: Shape count: 1
INFO: ROI 178:0 is a Polygon
INFO: Found 1 annotations for ROI: 179
INFO: Shape count: 1
INFO: ROI 179:0 is a Polygon
INFO: Found 1 annotations for ROI: 180
INFO: Shape count: 1
INFO: ROI 180:0 is a Polygon
INFO: Found 1 annotations for ROI: 181
INFO: Shape count: 1
INFO: ROI 181:0 is a Polygon
INFO: Found 1 annotations for ROI: 182
INFO: Shape count: 1
INFO: ROI 182:0 is a Polygon
INFO: Found 1 annotations for ROI: 183
INFO: Shape count: 1
INFO: ROI 183:0 is a Polygon
INFO: Found 1 annotations for ROI: 184
INFO: Shape count: 1
INFO: ROI 184:0 is a Polygon
INFO: Found 1 annotations for ROI: 185
INFO: Shape count: 1
INFO: ROI 185:0 is a Polygon
INFO: Found 1 annotations for ROI: 186
INFO: Shape count: 1
INFO: ROI 186:0 is a Polygon
INFO: Found 1 annotations for ROI: 187
INFO: Shape count: 1
INFO: ROI 187:0 is a Polygon
INFO: Found 1 annotations for ROI: 188
INFO: Shape count: 1
INFO: ROI 188:0 is a Polygon
INFO: Found 1 annotations for ROI: 189
INFO: Shape count: 1
INFO: ROI 189:0 is a Polygon
INFO: Found 1 annotations for ROI: 190
INFO: Shape count: 1
INFO: ROI 190:0 is a Polygon
INFO: Found 1 annotations for ROI: 191
INFO: Shape count: 1
INFO: ROI 191:0 is a Polygon
INFO: Found 1 annotations for ROI: 192
INFO: Shape count: 1
INFO: ROI 192:0 is a Polygon
INFO: Found 1 annotations for ROI: 193
INFO: Shape count: 1
INFO: ROI 193:0 is a Polygon
INFO: Found 1 annotations for ROI: 194
INFO: Shape count: 1
INFO: ROI 194:0 is a Polygon
INFO: Found 1 annotations for ROI: 195
INFO: Shape count: 1
INFO: ROI 195:0 is a Polygon
INFO: Found 1 annotations for ROI: 196
INFO: Shape count: 1
INFO: ROI 196:0 is a Polygon
INFO: Found 1 annotations for ROI: 197
INFO: Shape count: 1
INFO: ROI 197:0 is a Polygon
INFO: Found 1 annotations for ROI: 198
INFO: Shape count: 1
INFO: ROI 198:0 is a Polygon
INFO: Found 1 annotations for ROI: 199
INFO: Shape count: 1
INFO: ROI 199:0 is a Polygon
INFO: Found 1 annotations for ROI: 200
INFO: Shape count: 1
INFO: ROI 200:0 is a Polygon
INFO: Found 1 annotations for ROI: 201
INFO: Shape count: 1
INFO: ROI 201:0 is a Polygon
INFO: Found 1 annotations for ROI: 202
INFO: Shape count: 1
INFO: ROI 202:0 is a Polygon
INFO: Found 1 annotations for ROI: 203
INFO: Shape count: 1
INFO: ROI 203:0 is a Polygon
INFO: Found 1 annotations for ROI: 204
INFO: Shape count: 1
INFO: ROI 204:0 is a Polygon
INFO: Found 1 annotations for ROI: 205
INFO: Shape count: 1
INFO: ROI 205:0 is a Polygon
INFO: Found 1 annotations for ROI: 206
INFO: Shape count: 1
INFO: ROI 206:0 is a Polygon
INFO: Found 1 annotations for ROI: 207
INFO: Shape count: 1
INFO: ROI 207:0 is a Polygon
INFO: Found 1 annotations for ROI: 208
INFO: Shape count: 1
INFO: ROI 208:0 is a Polygon
INFO: Found 1 annotations for ROI: 209
INFO: Shape count: 1
INFO: ROI 209:0 is a Polygon
INFO: Found 1 annotations for ROI: 210
INFO: Shape count: 1
INFO: ROI 210:0 is a Polygon
INFO: Found 1 annotations for ROI: 211
INFO: Shape count: 1
INFO: ROI 211:0 is a Polygon
INFO: Found 1 annotations for ROI: 212
INFO: Shape count: 1
INFO: ROI 212:0 is a Polygon
INFO: Found 1 annotations for ROI: 213
INFO: Shape count: 1
INFO: ROI 213:0 is a Polygon
INFO: Found 1 annotations for ROI: 214
INFO: Shape count: 1
INFO: ROI 214:0 is a Polygon
INFO: Found 1 annotations for ROI: 215
INFO: Shape count: 1
INFO: ROI 215:0 is a Polygon
INFO: Found 1 annotations for ROI: 216
INFO: Shape count: 1
INFO: ROI 216:0 is a Polygon
INFO: Found 1 annotations for ROI: 217
INFO: Shape count: 1
INFO: ROI 217:0 is a Polygon
INFO: Found 1 annotations for ROI: 218
INFO: Shape count: 1
INFO: ROI 218:0 is a Polygon
INFO: Found 1 annotations for ROI: 219
INFO: Shape count: 1
INFO: ROI 219:0 is a Polygon
INFO: Found 1 annotations for ROI: 220
INFO: Shape count: 1
INFO: ROI 220:0 is a Polygon
INFO: Found 1 annotations for ROI: 221
INFO: Shape count: 1
INFO: ROI 221:0 is a Polygon
INFO: Found 1 annotations for ROI: 222
INFO: Shape count: 1
INFO: ROI 222:0 is a Polygon
INFO: Found 1 annotations for ROI: 223
INFO: Shape count: 1
INFO: ROI 223:0 is a Polygon
INFO: Found 1 annotations for ROI: 224
INFO: Shape count: 1
INFO: ROI 224:0 is a Polygon
INFO: Found 1 annotations for ROI: 225
INFO: Shape count: 1
INFO: ROI 225:0 is a Polygon
INFO: Found 1 annotations for ROI: 226
INFO: Shape count: 1
INFO: ROI 226:0 is a Polygon
INFO: Found 1 annotations for ROI: 227
INFO: Shape count: 1
INFO: ROI 227:0 is a Polygon
INFO: Found 1 annotations for ROI: 228
INFO: Shape count: 1
INFO: ROI 228:0 is a Polygon
INFO: Found 1 annotations for ROI: 229
INFO: Shape count: 1
INFO: ROI 229:0 is a Polygon
INFO: Found 1 annotations for ROI: 230
INFO: Shape count: 1
INFO: ROI 230:0 is a Polygon
INFO: Found 1 annotations for ROI: 231
INFO: Shape count: 1
INFO: ROI 231:0 is a Polygon
INFO: Found 1 annotations for ROI: 232
INFO: Shape count: 1
INFO: ROI 232:0 is a Polygon
INFO: Found 1 annotations for ROI: 233
ERROR: Error at line 127: null

ERROR: Script error
    at ome.xml.meta.OMEXMLMetadataImpl.getShapeCount(OMEXMLMetadataImpl.java:718)
    at ome.xml.meta.MetadataRetrieve$getShapeCount$3.call(Unknown Source)
    at Script3$_run_closure1.doCall(Script3.groovy:150)
    at jdk.internal.reflect.GeneratedMethodAccessor40.invoke(Unknown Source)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.base/java.lang.reflect.Method.invoke(Unknown Source)
    at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
    at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
    at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
    at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
    at groovy.lang.Closure.call(Closure.java:405)
    at groovy.lang.Closure.call(Closure.java:421)
    at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2330)
    at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2315)
    at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2356)
    at org.codehaus.groovy.runtime.dgm$186.invoke(Unknown Source)
    at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:244)
    at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:53)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
    at Script3.run(Script3.groovy:128)
    at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
    at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:798)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:732)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:712)
    at qupath.lib.gui.scripting.DefaultScriptEditor$2.run(DefaultScriptEditor.java:1128)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
    at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
    at java.base/java.lang.Thread.run(Unknown Source)
```

</details>

I created the OME-XML file I am trying to import here using the export script in this PR. I suspect that some annotations/ROIs were not recognized at the export step. Strangely, this is fine for the OME-XML plugin which imports the ROIs in this OME-XML file into OMERO, although there are some ``ignoring`` output lines, but no crash.

Attaching here the OME-XML too. 

[m6-melissa.txt](https://github.com/glencoesoftware/ome-omero-roitool/files/4552321/m6-melissa.txt)

**Comment by pwalczysko** [comment]
In summary,  this PR definitely brings a lot of improvement. 
The QP-m10 workflow is now made possible.

Regarding the (possible) magic wand tool problem https://github.com/glencoesoftware/ome-omero-roitool/pull/15#issuecomment-621252382 I can open a separate issue here, if this is better than fixing in present PR. Magic wand tool is definitely popular I guess. The problem si both the export script (not recognizing the magic wand ROI) and the import script (crashing fully just becuase there is a non-recognized ROI in the XML).

**Comment by melissalinkert** [comment]
Better to open a separate issue for the magic wand ROI issues, since they aren't caused by this PR.

**Comment by chris-allan** [comment]
An m10+ only workflow seems fine to me.

@melissalinkert: Is there any way for us to check the QuPath version from within the script and fail gracefully rather than spectacularly?

@pwalczysko: Are you up for checking things out on 0.2.0-m11 (released yesterday)?

**Comment by melissalinkert** [comment]
With cc598f4, the export script should fail immediately on any QuPath version prior to 0.2.0-m10.

**Comment by pwalczysko** [comment]
> @pwalczysko: Are you up for checking things out on 0.2.0-m11 (released yesterday)?

@chris-allan Will do, possibly today.

**Comment by chris-allan** [comment]
Great, thanks @pwalczysko!

I'll merge this since I want to make sure we have builds going up on GitHub releases for 5.4.x and 5.6.x and then if we see problems with 0.2.0-m11 we can address those separately.

**Comment by pwalczysko** [comment]
I can confirm that with QuPath 0.2.0-m11 on Mac OS I have

  - with the export script, the same experience as with m10 (works, and the subsequent import into OMERO works the same way too)
  - the errors with magic wand are unchanged between m10 and m11 in the import script
  - the simpler Annotation-ROIs work fine with the import script, just like with m10
  - the "Detection" ROIs export and import fine (as long as magic wand ROIs are absent), all like with m10

In summary, could not detect any m11 vs m10 differences

cc @chris-allan @melissalinkert

**Comment by chris-allan** [comment]
Thanks, @pwalczysko, much appreciated!
